tests/test_pylint_datetime.py

```diff
@@ -2,7 +2,308 @@
 # Copyright © 2023 Frequenz Energy-as-a-Service GmbH
 
 """Tests for the frequenz.pylint_datetime package."""
+from typing import Any
 
+import astroid  # type: ignore
+import pylint.testutils  # type: ignore
 
-def test_dummy() -> None:
-    """Test empty function."""
+from pylint_datetime import DatetimeChecker
+
+# tz=None and tzinfo=None is always allowed
+
+# it makes a difference for astroid if for example timedelta is called or datetime.timedelta
+# maybe this needs to be tested more thoroughly
+
+
+class TestDatetimeChecker(pylint.testutils.CheckerTestCase):  # type: ignore
+    """Test class for DatetimeChecker."""
+
+    CHECKER_CLASS = DatetimeChecker
+
+    def asserting_messages_calls(
+        self,
+        msg_id: str,
+        node: Any,
+        args: Any = None,
+    ) -> None:
+        """Assert messages for calls.
+
+        Args:
+            msg_id: message id.
+            node: node to check.
+            args: arguments to pass to the message.
+        """
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id=msg_id,
+                node=node,
+                args=args,
+            ),
+            ignore_position=True,
+        ):
+            self.checker.visit_call(node)
+
+    def asserting_messages_assign(
+        self,
+        msg_id: str,
+        node: Any,
+        node2: Any = None,
+        args: Any = None,
+    ) -> None:
+        """Assert messages for assignments.
+
+        Args:
+            msg_id: message id.
+            node: node to check.
+            node2: second node to check.
+            args: arguments to pass to the message.
+        """
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id=msg_id,
+                node=node if node2 is None else node2,
+                args=args,
+            ),
+            ignore_position=True,
+        ):
+            self.checker.visit_assign(node)
+
+    def test_timedelta_without_kargs(self) -> None:
+        """Test timedelta without keyword arguments."""
+        call_node_1, call_node_2 = astroid.extract_node(
+            """
+        import datetime
+        from datetime import timedelta
+        timedelta(1) #@
```

Comment:
What are these `#@`? Is a reference that then will be assigned to `call_node_1` and `call_node_2`?

Reply:
yes exactly, the #@ is the line that will then be assigned to the call nodes